dependencies/src/test/java/com/google/cloud/tools/opensource/dependencies/RepositoryUtilityTest.java

```diff
@@ -74,8 +77,25 @@ public void testReadBom_path() throws MavenRepositoryException, ArtifactDescript
     // However sometimes this list does change. If so, we want to 
     // output the specific difference so we can manually verify whether
     // the changes make sense. When they do make sense, we update the test. 
-    if (currentArtifacts.size() != 217) {
-      Truth.assertThat(currentArtifacts).containsExactlyElementsIn(oldArtifacts);
+    if (currentArtifacts.size() != 217) {      
+      Set<Artifact> current = Sets.newHashSet(currentArtifacts);
```

Comment:
This is a lot of code for something so simple. What about using [Expect](https://truth.dev/api/1.0/com/google/common/truth/Expect.html)?

```java
@Rule public final Expect expect = Expect.create();

...

Set<Artifact> current = Sets.newHashSet(currentArtifacts);
Set<Artifact> old = Sets.newHashSet(oldArtifacts);
expect.withMessage("added artifacts").that(Sets.difference(current, old)).isEmpty();
expect.WithMessage("removed artifacts").that(Sets.difference(old, current)).isEmpty();
```

Reply:
I don't think that will show us which artifacts were added and removed though.

Reply:
It will! `assertThat(foo).isEmpty()` shows the unexpected elements on failure. That's part of the design principle of Truth: it tries to show as much useful information as possible.

`assertTrue(foo.isEmpty())` can't do that.

Reply:
will it show both or does it fail on the first?

Reply:
It will show both. (Somehow my response appeared above your question.)